src/Parser/Mage/Fire/Modules/Features/Checklist.js

```diff
@@ -0,0 +1,178 @@
+      requirements: () => {
+        return [
+          new Requirement({
+            name: "Combustion Used with < 2 Phoenix Flames Charges",
+            check: () => this.combustionCharges.phoenixFlamesThresholds,
+          }),
+          new Requirement({
+            name: `Combustion Used with < ${this.combatants.selected.hasTalent(SPELLS.FLAME_ON_TALENT.id) ? 2 : 1} Fire Blast Charges`,
+            check: () => this.combustionCharges.fireBlastThresholds,
+          }),
+          new Requirement({
+            name: "Spells hard cast while instants were available",
+            check: () => this.combustionSpellUsage.suggestionThresholds,
+          }),
+        ];
+      },
+    }),
+    new Rule({
+      name: 'Use your cooldowns',
+      description: <Wrapper>Your cooldowns are a major contributor to your DPS, and should be used as frequently as possible throughout a fight. A cooldown should be held on to only if a priority DPS phase is coming <em>soon</em>. Holding cooldowns too long will hurt your DPS.</Wrapper>,
+      requirements: () => {
+        const combatant = this.combatants.selected;
+        return [
+          new GenericCastEfficiencyRequirement({
+            spell: SPELLS.COMBUSTION,
+          }),
+          new GenericCastEfficiencyRequirement({
+            spell: SPELLS.CINDERSTORM_TALENT,
+            when: combatant.hasTalent(SPELLS.CINDERSTORM_TALENT.id),
+          }),
+          new GenericCastEfficiencyRequirement({
+            spell: SPELLS.MIRROR_IMAGE_TALENT,
+            when: combatant.hasTalent(SPELLS.MIRROR_IMAGE_TALENT.id),
+          }),
+          new GenericCastEfficiencyRequirement({
+            spell: SPELLS.RUNE_OF_POWER_TALENT,
+            when: combatant.hasTalent(SPELLS.RUNE_OF_POWER_TALENT.id),
+          }),
+          new GenericCastEfficiencyRequirement({
+            spell: SPELLS.BLAST_WAVE_TALENT,
+            when: combatant.hasTalent(SPELLS.BLAST_WAVE_TALENT.id),
+          }),
+          new GenericCastEfficiencyRequirement({
+            spell: SPELLS.METEOR_TALENT,
+            when: combatant.hasTalent(SPELLS.METEOR_TALENT.id),
+          }),
+        ];
+      },
+    }),
+
+    new Rule({
+      name: 'Utilize your talents',
```

Comment:
Is it possible to hide this if talent choices make there be no Requirements?

Reply:
i have no idea ... is there someone else doing something like that? I can try to figure it out.

Reply:
The only thing that comes into mind is wrapping it somehow in a ternary operator, or providing another prop `when` to the Rule class. Might be a good thing to add sometimes as having an empty Rule is confusing.